src/Core/Content/Seo/SeoUrlGenerator.php

```diff
@@ -63,19 +75,23 @@ public function generate(array $ids, string $template, SeoUrlRouteInterface $rou
             return new RepositoryIterator($repository, $context, $criteria);
         });
 
-        $this->setTwigTemplate($config, $template);
+        $twig = $this->createTwigTemplate($config, $template);
 
-        while ($entities = $iterator->fetch()) {
-            yield from $this->generateUrls($route, $config, $salesChannel, $entities);
+        if ($twig instanceof Environment) {
```

Comment:
why should be twig something different

Reply:
Read the factory code/type hint. It can be null when setting of the template fails.